doc/source/train/examples/hpu/bert.ipynb

```diff
@@ -0,0 +1,527 @@
+    "    eval_dataset = dataset[\"test\"].select(range(1000)).map(tokenize_function, batched=True)\n",
+    "\n",
+    "    # Prepare dataloader for each worker\n",
+    "    dataloaders = {}\n",
+    "    dataloaders[\"train\"] = torch.utils.data.DataLoader(\n",
+    "        train_dataset, \n",
+    "        shuffle=True, \n",
+    "        collate_fn=transformers.default_data_collator, \n",
+    "        batch_size=batch_size\n",
+    "    )\n",
+    "    dataloaders[\"test\"] = torch.utils.data.DataLoader(\n",
+    "        eval_dataset, \n",
+    "        shuffle=True, \n",
+    "        collate_fn=transformers.default_data_collator, \n",
+    "        batch_size=batch_size\n",
+    "    )\n",
+    "\n",
+    "    # Model\n",
+    "    model = AutoModelForSequenceClassification.from_pretrained(\n",
+    "        \"bert-base-cased\", num_labels=5\n",
+    "    )\n",
+    "\n",
+    "    device = ray.train.torch.get_device()\n",
+    "    model = model.to(device)\n",
+    "    \n",
+    "    # \n",
+    "    optimizer = torch.optim.SGD(model.parameters(), lr=lr, momentum=0.9)\n",
+    "\n",
+    "    # Start training loops\n",
+    "    for epoch in range(epochs):\n",
+    "        # Each epoch has a training and validation phase\n",
+    "        for phase in [\"train\", \"test\"]:\n",
+    "            if phase == \"train\":\n",
+    "                model.train()  # Set model to training mode\n",
+    "            else:\n",
+    "                model.eval()  # Set model to evaluate mode\n",
+    "\n",
+    "            # breakpoint()\n",
+    "            for batch  in dataloaders[phase]:\n",
+    "                batch = {k: v.to(device) for k, v in batch.items()}\n",
+    "\n",
+    "                # zero the parameter gradients\n",
+    "                optimizer.zero_grad()\n",
+    "\n",
+    "                # forward\n",
+    "                with torch.set_grad_enabled(phase == \"train\"):\n",
+    "                    # Get model outputs and calculate loss\n",
+    "                    \n",
+    "                    outputs = model(**batch)\n",
+    "                    loss = outputs.loss\n",
+    "\n",
+    "                    # backward + optimize only if in training phase\n",
+    "                    if phase == \"train\":\n",
+    "                        loss.backward()\n",
+    "                        optimizer.step()\n",
+    "                        print(f\"train epoch:[{epoch}]\\tloss:{loss:.6f}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Main Training Function\n",
+    "\n",
+    "The `train_bert` function sets up the distributed training environment using Ray and starts the training process."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def train_bert(num_workers=2, use_gpu=False):\n",
```

Comment:
We are not using the `use_gpu` flag here? We can focus on HPU usage and just remove this option.